File: src/mod3_oop/lesson26_constructors/OOPDemo.java
package mod3_oop.lesson26_constructors;

public class OOPDemo {
    public static void main(String[] args) {

        // java
        Person konstantin = new Person();
        konstantin.name = "Konstantin";
        konstantin.age = 20;
        konstantin.sayHello();

        Person ivan = new Person("Ivan");
        ivan.sayHello();

        Person john = new Person("John Smith", 38);
        john.sayHello();

        // kotlin
        PersonKt ktPerson = new PersonKt();
        ktPerson.setName("Konstantin");
        ktPerson.setAge(34);
        ktPerson.sayHello();

        var selena = new PersonKt("Selena Gomez", 32);
        selena.sayHello();

    }
}

File: src/mod3_oop/lesson32_deep_copy/OOPDemo.kt
package mod3_oop.lesson32_deep_copy

fun main() {
    val selena = PersonKt.getInstance("Selena Gomez", 38)
    selena.sayHello()
    println(selena.age)

    val madonna = PersonKt.getInstance("Madonna")
    // val thirdPerson = selena // by reference
    val thirdPerson = PersonKt.getInstance(selena) // deep copy
    thirdPerson.name = "Alexa Vega"
    println(selena.name)
}